src/platform/plugins/shared/discover/public/application/main/components/layout/cascaded_documents/blocks/cascade_leaf_component.tsx

```diff
@@ -0,0 +1,304 @@
+    position: 'relative',
+    scrollbarWidth: 'thin',
+  }),
+  virtualizerInner: css({
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    width: '100%',
+    overflowAnchor: 'none',
+  }),
+  displayFlex: css({ display: 'flex' }),
+};
+
+/**
+ * A custom grid body implementation for the unified data table to be used in the cascade leaf cells
+ * that allows for nested cascade virtualization that's compatible with the EUI Data Grid
+ */
+export const CustomCascadeGridBodyMemoized = React.memo(function CustomCascadeGridBody({
+  isFullScreenMode,
+  initialOffset,
+  data,
+  getScrollElement,
+  getScrollMargin,
+  Cell,
+  visibleColumns,
+  visibleRowData,
+  headerRow,
+  footerRow,
+  setCustomGridBodyProps,
+}: CustomCascadeGridBodyProps) {
+  const visibleRows = useMemo(
+    () => data.slice(visibleRowData.startRow, visibleRowData.endRow),
+    [data, visibleRowData.startRow, visibleRowData.endRow]
+  );
+  const customGridBodyScrollContainerRef = useRef<HTMLDivElement | null>(null);
+
+  const scrollMargin = useMemo(() => getScrollMargin(), [getScrollMargin]);
+
+  // create scroll element reference for custom nested virtualized grid
+  const virtualizerScrollElementRef = useRef<Element | null>(getScrollElement());
+
+  const scrollElementGetter = useCallback(() => virtualizerScrollElementRef.current, []);
+
+  useEffect(() => {
+    const defaultScrollElement = getScrollElement();
+
+    if (
+      isFullScreenMode &&
+      customGridBodyScrollContainerRef.current &&
+      !virtualizerScrollElementRef.current?.isSameNode(customGridBodyScrollContainerRef.current)
+    ) {
+      // assign the custom grid body element as scrollable element in full screen mode
+      // customCascadeGridScrollRef.current = $gridBody;
+      virtualizerScrollElementRef.current = customGridBodyScrollContainerRef.current;
+    } else if (
+      !isFullScreenMode &&
+      !virtualizerScrollElementRef.current?.isSameNode(defaultScrollElement)
+    ) {
+      // when exiting full screen mode we ensure the scroll element is reset to the default scroll container
+      virtualizerScrollElementRef.current = defaultScrollElement;
+    }
+  }, [getScrollElement, isFullScreenMode]);
+
+  const virtualizer = useVirtualizer({
+    count: visibleRows.length,
+    estimateSize: () => 50,
+    overscan: 15,
+    initialOffset,
+    scrollMargin,
+    getScrollElement: scrollElementGetter,
+    useAnimationFrameWithResizeObserver: true,
+  });
+
+  const items = virtualizer.getVirtualItems();
+
+  return (
+    <div
+      data-test-subj="discoverCascadeCustomDataGridBody"
+      role="rowgroup"
+      css={customCascadeGridBodyStyle.wrapper}
+    >
+      <div css={customCascadeGridBodyStyle.headerRow}>{headerRow}</div>
+      <div
+        ref={customGridBodyScrollContainerRef}
+        css={customCascadeGridBodyStyle.virtualizerContainer}
+      >
+        <div
+          style={{
+            height: virtualizer.getTotalSize(),
+          }}
```

Comment:
Can we use `css` instead of `style` prop?

Reply:
There's a consensus to use the `style` property instead of `css` for dynamic values with emotion, [see](https://github.com/elastic/eui/discussions/6828#discussioncomment-12867545)